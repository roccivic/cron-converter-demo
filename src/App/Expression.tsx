import { useDispatch } from "react-redux";
import { SET_EXPRESSION } from "../redux/slice";
import { useAppSelector } from "../redux/store";

export const Expression = () => {
  const dispatch = useDispatch();
  const value = useAppSelector((state) => state.expression);
  return (
    <div className="form-floating" style={{ marginBottom: 15 }}>
      <input
        type="email"
        className="form-control"
        id="expression"
        onChange={(e) => dispatch(SET_EXPRESSION(e.target.value))}
        value={value}
      />
      <label htmlFor="expression">Cron expression</label>
    </div>
  );
};
