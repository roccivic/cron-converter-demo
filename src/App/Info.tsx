import { FaInfoCircle } from "react-icons/fa";

export const Info = () => 
<div className="alert alert-info">
  <FaInfoCircle /> Type in an expression to select the values in the <code>select</code>s
  or pick the values in the <code>select</code>s to get an expression.
</div>