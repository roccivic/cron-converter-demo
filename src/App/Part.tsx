import { SET_VALUE } from "../redux/slice";
import { useDispatch } from "react-redux";
import { useAppSelector } from "../redux/store";

type PartProps = {
  id: string;
  min: number;
  max: number;
  index: number;
};

const range = (min: number, max: number) => {
  const arr: number[] = [];
  for (let i = min; i <= max; i++) {
    arr.push(i);
  }
  return arr;
};

export const Part = ({ id, min, max, index }: PartProps) => {
  const dispatch = useDispatch();
  const options = range(min, max);
  const value = useAppSelector((state) => state.array[index].map(String));
  return (
    <div style={{ width: "18%", marginLeft: "1%", marginRight: "1%" }}>
      <label htmlFor={id} style={{ textTransform: "capitalize" }}>
        <strong>{id}</strong>
      </label>
      <select
        id={id}
        className="form-control"
        multiple={true}
        value={value}
        size={60}
        onChange={(e) =>
          dispatch(
            SET_VALUE({
              index,
              values: [...e.target.options].filter((o) => o.selected).map((o) => Number(o.value)),
            })
          )
        }
      >
        {options.map((value) => (
          <option key={value} value={value}>
            {value}
          </option>
        ))}
      </select>
    </div>
  );
};
