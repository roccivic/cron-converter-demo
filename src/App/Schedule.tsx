import { useAppSelector } from "../redux/store";
import { FaClock, FaExclamationCircle } from "react-icons/fa";

export const Schedule = () => {
  const error = useAppSelector((state) => state.error);
  const next = useAppSelector((state) => state.next);
  const prev = useAppSelector((state) => state.prev);
  if (error) {
    return (
      <div className="alert alert-danger">
        <FaExclamationCircle /> {error}
      </div>
    );
  } else {
    return (
      <div className="alert alert-success">
        <div>
          <FaClock /> Would execute next at {next}
        </div>
        <div>
          <FaClock /> Would have last executed at {prev}
        </div>
      </div>
    );
  }
};
