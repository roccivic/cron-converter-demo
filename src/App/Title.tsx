export const Title = () => (
  <div
    style={{
      padding: 20,
      background: "#333",
      color: "white",
      fontSize: 22,
    }}
  >
    <div className="container">
      <span className="code">cron-converter demo</span>
    </div>
  </div>
);
