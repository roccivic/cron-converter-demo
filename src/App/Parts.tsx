import { Part } from "./Part";
import { getUnits } from "cron-converter";

export const Parts = () => (
  <div style={{ display: "flex" }}>
    {getUnits().map((unit, index) => (
      <Part key={unit.name} id={unit.name} index={index} min={unit.min} max={unit.max} />
    ))}
  </div>
);
