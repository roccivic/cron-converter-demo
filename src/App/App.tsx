import { Expression } from "./Expression";
import { Info } from "./Info";
import { Parts } from "./Parts";
import { Provider } from "react-redux";
import { Schedule } from "./Schedule";
import { store } from "../redux/store";
import { Title } from "./Title";
import { FaExternalLinkAlt } from "react-icons/fa";

export const App = () => (
  <div>
    <Title />
    <div className="container" style={{ padding: 15 }}>
      <p>
        <a href="https://github.com/roccivic/cron-converter">
          <FaExternalLinkAlt style={{ fontSize: 12, verticalAlign: -1 }} /> View the cron-converter
          repository on GitHub.
        </a>
      </p>
      <Provider store={store}>
        <div className="row">
          <div className="col-xl-6">
            <Expression />
          </div>
          <div className="col-xl-6">
            <Schedule />
          </div>
        </div>
        <Info />
        <Parts />
      </Provider>
    </div>
  </div>
);
