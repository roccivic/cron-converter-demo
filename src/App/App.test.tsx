import { render } from "@testing-library/react";
import { App } from "./App";

test("<App />", () => {
  const comp = render(<App />);
  expect(comp.container).toMatchSnapshot();
});
