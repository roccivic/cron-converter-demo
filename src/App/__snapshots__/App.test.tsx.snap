// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> 1`] = `
<div>
  <div>
    <div
      style="padding: 20px; background: rgb(51, 51, 51); color: white; font-size: 22px;"
    >
      <div
        class="container"
      >
        <span
          class="code"
        >
          cron-converter demo
        </span>
      </div>
    </div>
    <div
      class="container"
      style="padding: 15px;"
    >
      <p>
        <a
          href="https://github.com/roccivic/cron-converter"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="font-size: 12px; vertical-align: -1px;"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
            />
          </svg>
           View the cron-converter repository on GitHub.
        </a>
      </p>
      <div
        class="row"
      >
        <div
          class="col-xl-6"
        >
          <div
            class="form-floating"
            style="margin-bottom: 15px;"
          >
            <input
              class="form-control"
              id="expression"
              type="email"
              value="*/15 0-11 1,10,20 * *"
            />
            <label
              for="expression"
            >
              Cron expression
            </label>
          </div>
        </div>
        <div
          class="col-xl-6"
        >
          <div
            class="alert alert-success"
          >
            <div>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
                />
              </svg>
               Would execute next at 
              November 20, 2022 at 12:00 AM EST
            </div>
            <div>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
                />
              </svg>
               Would have last executed at 
              November 10, 2022 at 11:45 AM EST
            </div>
          </div>
        </div>
      </div>
      <div
        class="alert alert-info"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
          />
        </svg>
         Type in an expression to select the values in the 
        <code>
          select
        </code>
        s or pick the values in the 
        <code>
          select
        </code>
        s to get an expression.
      </div>
      <div
        style="display: flex;"
      >
        <div
          style="width: 18%; margin-left: 1%; margin-right: 1%;"
        >
          <label
            for="minute"
            style="text-transform: capitalize;"
          >
            <strong>
              minute
            </strong>
          </label>
          <select
            class="form-control"
            id="minute"
            multiple=""
            size="60"
          >
            <option
              value="0"
            >
              0
            </option>
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="11"
            >
              11
            </option>
            <option
              value="12"
            >
              12
            </option>
            <option
              value="13"
            >
              13
            </option>
            <option
              value="14"
            >
              14
            </option>
            <option
              value="15"
            >
              15
            </option>
            <option
              value="16"
            >
              16
            </option>
            <option
              value="17"
            >
              17
            </option>
            <option
              value="18"
            >
              18
            </option>
            <option
              value="19"
            >
              19
            </option>
            <option
              value="20"
            >
              20
            </option>
            <option
              value="21"
            >
              21
            </option>
            <option
              value="22"
            >
              22
            </option>
            <option
              value="23"
            >
              23
            </option>
            <option
              value="24"
            >
              24
            </option>
            <option
              value="25"
            >
              25
            </option>
            <option
              value="26"
            >
              26
            </option>
            <option
              value="27"
            >
              27
            </option>
            <option
              value="28"
            >
              28
            </option>
            <option
              value="29"
            >
              29
            </option>
            <option
              value="30"
            >
              30
            </option>
            <option
              value="31"
            >
              31
            </option>
            <option
              value="32"
            >
              32
            </option>
            <option
              value="33"
            >
              33
            </option>
            <option
              value="34"
            >
              34
            </option>
            <option
              value="35"
            >
              35
            </option>
            <option
              value="36"
            >
              36
            </option>
            <option
              value="37"
            >
              37
            </option>
            <option
              value="38"
            >
              38
            </option>
            <option
              value="39"
            >
              39
            </option>
            <option
              value="40"
            >
              40
            </option>
            <option
              value="41"
            >
              41
            </option>
            <option
              value="42"
            >
              42
            </option>
            <option
              value="43"
            >
              43
            </option>
            <option
              value="44"
            >
              44
            </option>
            <option
              value="45"
            >
              45
            </option>
            <option
              value="46"
            >
              46
            </option>
            <option
              value="47"
            >
              47
            </option>
            <option
              value="48"
            >
              48
            </option>
            <option
              value="49"
            >
              49
            </option>
            <option
              value="50"
            >
              50
            </option>
            <option
              value="51"
            >
              51
            </option>
            <option
              value="52"
            >
              52
            </option>
            <option
              value="53"
            >
              53
            </option>
            <option
              value="54"
            >
              54
            </option>
            <option
              value="55"
            >
              55
            </option>
            <option
              value="56"
            >
              56
            </option>
            <option
              value="57"
            >
              57
            </option>
            <option
              value="58"
            >
              58
            </option>
            <option
              value="59"
            >
              59
            </option>
          </select>
        </div>
        <div
          style="width: 18%; margin-left: 1%; margin-right: 1%;"
        >
          <label
            for="hour"
            style="text-transform: capitalize;"
          >
            <strong>
              hour
            </strong>
          </label>
          <select
            class="form-control"
            id="hour"
            multiple=""
            size="60"
          >
            <option
              value="0"
            >
              0
            </option>
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="11"
            >
              11
            </option>
            <option
              value="12"
            >
              12
            </option>
            <option
              value="13"
            >
              13
            </option>
            <option
              value="14"
            >
              14
            </option>
            <option
              value="15"
            >
              15
            </option>
            <option
              value="16"
            >
              16
            </option>
            <option
              value="17"
            >
              17
            </option>
            <option
              value="18"
            >
              18
            </option>
            <option
              value="19"
            >
              19
            </option>
            <option
              value="20"
            >
              20
            </option>
            <option
              value="21"
            >
              21
            </option>
            <option
              value="22"
            >
              22
            </option>
            <option
              value="23"
            >
              23
            </option>
          </select>
        </div>
        <div
          style="width: 18%; margin-left: 1%; margin-right: 1%;"
        >
          <label
            for="day"
            style="text-transform: capitalize;"
          >
            <strong>
              day
            </strong>
          </label>
          <select
            class="form-control"
            id="day"
            multiple=""
            size="60"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="11"
            >
              11
            </option>
            <option
              value="12"
            >
              12
            </option>
            <option
              value="13"
            >
              13
            </option>
            <option
              value="14"
            >
              14
            </option>
            <option
              value="15"
            >
              15
            </option>
            <option
              value="16"
            >
              16
            </option>
            <option
              value="17"
            >
              17
            </option>
            <option
              value="18"
            >
              18
            </option>
            <option
              value="19"
            >
              19
            </option>
            <option
              value="20"
            >
              20
            </option>
            <option
              value="21"
            >
              21
            </option>
            <option
              value="22"
            >
              22
            </option>
            <option
              value="23"
            >
              23
            </option>
            <option
              value="24"
            >
              24
            </option>
            <option
              value="25"
            >
              25
            </option>
            <option
              value="26"
            >
              26
            </option>
            <option
              value="27"
            >
              27
            </option>
            <option
              value="28"
            >
              28
            </option>
            <option
              value="29"
            >
              29
            </option>
            <option
              value="30"
            >
              30
            </option>
            <option
              value="31"
            >
              31
            </option>
          </select>
        </div>
        <div
          style="width: 18%; margin-left: 1%; margin-right: 1%;"
        >
          <label
            for="month"
            style="text-transform: capitalize;"
          >
            <strong>
              month
            </strong>
          </label>
          <select
            class="form-control"
            id="month"
            multiple=""
            size="60"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="11"
            >
              11
            </option>
            <option
              value="12"
            >
              12
            </option>
          </select>
        </div>
        <div
          style="width: 18%; margin-left: 1%; margin-right: 1%;"
        >
          <label
            for="weekday"
            style="text-transform: capitalize;"
          >
            <strong>
              weekday
            </strong>
          </label>
          <select
            class="form-control"
            id="weekday"
            multiple=""
            size="60"
          >
            <option
              value="0"
            >
              0
            </option>
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</div>
`;
